integration/test/performance_tracker/performance_query_utils.go

```diff
@@ -0,0 +1,143 @@
+	}
+	
+	timeNow := time.Now()
+	input := &cloudwatch.GetMetricDataInput{
+		EndTime:   aws.Time(time.Unix(timeNow.Unix(), 0)),
+		StartTime: aws.Time(time.Unix(timeNow.Add(time.Duration(-timeDiff)*time.Minute).Unix(), 0)),
+		MetricDataQueries: metricDataQueries,
+	}
+
+	return input, nil
+}
+
+// ConstructMetricDataQuery is a helper function for GenerateGetMetricInputStruct and constructs individual metric requests
+func ConstructMetricDataQuery(id, namespace, dimensionName, dimensionValue, metricName string, timeDiff int) (types.MetricDataQuery) {
+	query := types.MetricDataQuery{
+		Id: aws.String(id),
+		MetricStat: &types.MetricStat{
+			Metric: &types.Metric{
+				Namespace:  aws.String(namespace),
+				MetricName: aws.String(metricName),
+				Dimensions: []types.Dimension{
+					types.Dimension{
+						Name:  aws.String(dimensionName),
+						Value: aws.String(dimensionValue),
+					},
+				},
+			},
+			Period: aws.Int32(int32(Period)),
+			Stat:   aws.String(Stat),
+		},
+	}
+
+	return query
+}
+
+func GetPerformanceMetrics(instanceId string, agentRuntime int) ([]byte, error) {
+	agentContext := context.TODO()
+
+	//load default configuration
+	cfg, err := config.LoadDefaultConfig(agentContext)
+	if err != nil {
+		return nil, err
+	}
+
+	client := cloudwatch.NewFromConfig(cfg)
+
+	//get metric measurements from config file
+	file, err := os.ReadFile("./resources/config.json")
+	if err != nil {
+		return nil, err
+	}
+
+	var cfgFileData map[string]interface{}
+	err = json.Unmarshal(file, &cfgFileData)
+	if err != nil {
+		return nil, err
+	}
+
+	//go through the config json to get to the procstat metrics
+	procstatList := cfgFileData["metrics"].(map[string]interface{})["metrics_collected"].(map[string]interface{})["procstat"].([]interface{})
+	
+	//within procstat metrics, find cloudwatch-agent process
+	cloudwatchIndex := -1
+	for i, process := range procstatList {
+		if process.(map[string]interface{})["exe"].(string) == "cloudwatch-agent" {
+			cloudwatchIndex = i
+		}
+	}
+
+	//check to see if the process was not found
+	if  cloudwatchIndex == -1 {
+		return nil, errors.New("cloudwatch-agent process not found in cloudwatch agent config")
+	}
+
+	//use the index to get the rest of the path
+	metricList := procstatList[cloudwatchIndex].(map[string]interface{})["measurement"].([]interface{})
+
+	//convert the resulting []interface{} to []string and create matching metric ids for each one
+	metricNames := make([]string, len(metricList))
+	ids := make([]string, len(metricList))
+	for i, metricName := range metricList {
+		metricNames[i] = "procstat_" + metricName.(string)
+		ids[i] = fmt.Sprint("m", i + 1)
+	}
+
+	//make input struct
+	input, err := GenerateGetMetricInputStruct(ids, metricNames, instanceId, agentRuntime)
+	if err != nil {
+		return nil, err
+	}
+
+	//call to cloudwatch agent API
```

Comment:
nitpick, the agent doesn't have an API. The agent is just a normal client of CloudWatch like everyone else
```suggestion
	//call to CloudWatch API
```